packages/celotool/src/lib/helm_deploy.ts

```diff
@@ -701,7 +701,7 @@ export async function resetAndUpgradeHelmChart(celoEnv: string, useExistingGenes
   await scaleResource(celoEnv, 'StatefulSet', privateTxNodesSetname, 0, true)
   await scaleResource(celoEnv, 'Deployment', bootnodeName, 0)
 
-  await deletePersistentVolumeClaims(celoEnv)
+  await deletePersistentVolumeClaims(celoEnv, persistentVolumeClaimsLabels)
   await sleep(10000)
```

Comment:
I imagine this is here to ensure it's deleted prior to upgrade. Is it possible to query the state of the volumeClaims explicitly before continuing?

Reply:
Maybe it could be check but not sure if we need this or add any extra security.
There is a double abstraction layer for volumes in Kuberntes. The `PersistentVolumeClaim` represents a request for a volume. The `PersistentVolume` object represents a mapping between the physical volume and can be created automatically when creating a PVC resource.
Depending on some configuration on the `PersistentVolume` the physical volume will be deleted or not if the pvc is removed (for the storage class we are using here it will be deleted if nobody changes it manually).
The good part here of this abstraction is that recreating the PVC also force to create a new PV resource (with different name), so avoid any collision. Also the kubernetes controller should be responsible of deleting the old pv and azure disk.